src/compute.ts

```diff
@@ -13,50 +13,79 @@
 // limitations under the License.
 
 import * as aws from "@pulumi/aws";
+
 import { EnforcementLevel, ResourceValidationPolicy, validateTypedResource } from "@pulumi/policy";
 
-export const compute: ResourceValidationPolicy[] = [
-    ec2InstanceDetailedMonitoringEnabled("mandatory"),
-    ec2InstanceNoPublicIP("mandatory"),
-    ec2VolumeInUseCheck("mandatory", true),
-    elbAccessLoggingEnabled("mandatory"),
-    encryptedVolumes("mandatory"),
-];
+import { registerPolicy } from "./awsGuard";
+import { defaultEnforcementLevel } from "./enforcementLevel";
+import { PolicyArgs } from "./policyArgs";
+import { getValueOrDefault } from "./util";
+
+// Mixin additional properties onto AwsGuardArgs.
+declare module "./awsGuard" {
+    interface AwsGuardArgs {
+        ec2InstanceDetailedMonitoringEnabled?: EnforcementLevel;
+        ec2InstanceNoPublicIP?: EnforcementLevel;
+        ec2VolumeInUseCheck?: EnforcementLevel | Ec2VolumeInUseCheckArgs;
+        elbAccessLoggingEnabled?: EnforcementLevel;
+        encryptedVolumes?: EnforcementLevel | EncryptedVolumesArgs;
+    }
+}
```

Comment:
I considered having separate interfaces for each policy, e.g.:

```typescript
export interface Ec2InstanceDetailedMonitoringEnabledPolicyPackArgs {
    ec2InstanceDetailedMonitoringEnabled?: EnforcementLevel;
}

export interface Ec2InstanceNoPublicIPPolicyPackArgs {
    ec2InstanceNoPublicIP?: EnforcementLevel;
}

// etc.
```

And then defining `AwsGuardArgs` as:

```typescript
export interface AwsGuardArgs extends
    Ec2InstanceDetailedMonitoringEnabledPolicyPackArgs,
    Ec2InstanceNoPublicIPPolicyPackArgs,
    // etc.
    {
    all?: EnforcementLevel;
}
```

Or:

```typescript
export type AwsGuardArgs =
    Ec2InstanceDetailedMonitoringEnabledPolicyPackArgs & 
    Ec2InstanceNoPublicIPPolicyPackArgs &
    // etc.
```

But then it adds a bunch of unnecessary types and forces you to add the definitions where `AwsGuardArgs` is declared.

Doing it the way I've done it is consistent with how we apply "mixins" elsewhere, for example: https://github.com/pulumi/pulumi-aws/blob/daa4d0d4ea9ca5c785f2673f2c87c1865473a506/sdk/nodejs/s3/s3Mixins.ts#L215-L246